raft/src/test/java/org/apache/kafka/raft/DistributedCounter.java

```diff
@@ -72,8 +81,9 @@ public synchronized void apply(Records records) {
             if (!batch.isControlBatch()) {
                 for (Record record : batch) {
                     int value = deserialize(record);
-                    if (value != committed.get() + 1) {
-                        throw new IllegalStateException("Detected invalid increment in record at offset " + record.offset() +
+
+                    if (value != committed.get() && value != committed.get() + 1) {
```

Comment:
With the new appender interface, we no longer use `uncommitted` to buffer data, so it is possible to see repetitive increment without an actual increase on the sequence.

Reply:
Is it worth to add this as a comment?

Reply:
Seems like a regression in behavior. This state machine will be much more valuable in testing if we can keep its current semantics. Can we not still achieve an atomic increment?